apps/molgenis-components/src/components/forms/RowEdit.vue

```diff
@@ -163,40 +165,49 @@ export default {
         }
       });
     },
-    //create a filter in case inputs are linked by overlapping refs
-    refLinkFilter(column: IColumn) {
-      //need to figure out what refs overlap
-      if (
-        column.refLinkId &&
-        this.showColumn(column) &&
-        this.internalValues[column.refLinkId]
-      ) {
-        let filter: Record<string, any> = {};
-        this.tableMetaData.columns.forEach((column2: IColumn) => {
-          if (column2.id === column.refLinkId) {
-            this.schemaMetaData.tables.forEach((table: ITableMetaData) => {
-              //check how the refTableId overlaps with columns in our column
-              if (table.id === column.refTableId) {
-                table.columns.forEach((column3) => {
-                  if (
-                    column3.key === 1 &&
-                    column3.refTableId === column2.refTableId
-                  ) {
-                    filter[column3.id] = {
+    //update reflink filters and reset values if reflink changes
+    async updateRefLinks(changedColumn: IColumn) {
+      //need to figure out what refs overlap with the changed column
+      for (const column1 of this.tableMetaData.columns) {
+        if (
+          this.internalValues[changedColumn.id] &&
+          column1.refLinkId == changedColumn.id &&
+          this.showColumn(column1)
+        ) {
+          console.log("check");
+          //check how the refTableId overlaps with columns in our column
+          for (const table of this.schemaMetaData.tables) {
+            if (table.id === column1.refTableId) {
+              console.log("check2");
+              for (const column2 of table.columns) {
```

Comment:
same as with column1, maybe in the spirit of `refTableColumn`.

Reply:
ditto